modules/presentation.py

```diff
@@ -0,0 +1,215 @@
+# Presentation file
+# Contains charts
+# IMPORTS:
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import re
+import nltk
+import datetime
+
+
+"""
+How to use:
+
+You have to have an array of tweets.
+1. Sort
+2. Get sentiment
+3. Plot
+4. Save plot or show
+
+First, you sort the tweets, using for example get_by_key_value or get_tweets_in_daterange
+Then you use get_sentiment 
+Then you feed the result of get_sentiment to a plotter function
+
+You can chain several filters.
+
+"""
+
+# Change here, if sentiments change
+sentiments = ["Positive", "Negative", "Uncertain"]
+
+# 1 SORTING
+# FILTERING / SORTING FUNCTIONS START
+def get_tweets_in_daterange(tweets, start_date, end_date):
+    """
+    Category: Filter function
+
+    Returns all tweets with "date" between start_date and end_date
+
+    Parameters:\n
+        tweets: array of tweets
+        start_date and end_date are datetime objects
+
+    Returns:\n
+        Filtered list of tweets within start_date and end_date
+    """
+    if not isinstance(start_date, datetime.date) or not isinstance(end_date, datetime.date):
+        raise Exception("Wrong date format. Has to be datetime.date")
+
+    if start_date > end_date:
+        raise Exception("Start_date was before end_date.")
+
+    return list(
+        filter(
+            lambda tweet: tweet["date"] <= end_date and tweet["date"] >= start_date,
+            tweets,
+        )
+    )
+
+
+def get_by_key_value(tweets, key, value):
+    """
+    Category: Filter function
+
+    Get tweets from tweet array, by key and value. 
+
+    For example:\n
+        key = hashtags
+        value = #Biden
+
+    Parameters: \n
+        tweets = Array of tweets you want to filter
+        key = String: one of: [hashtags, people, urls]
+        value = String
+
+    Returns: \n
+        Filtered List
+    """
+    if key is not in ["hashtags", "people", "urls"]:
+        raise Exception('key has to be "hashtags", "people", "urls"')
+
+    return list(filter(lambda tweet: value in tweet[key], tweets))
+
+
+def get_by_sentiment(tweets, sentiment):
+    f"""
```

Comment:
What does `f` mean?

Reply:
https://realpython.com/python-f-strings/

Reply:
Instead of .format, you just put f in front of the string, and you can use { } to put in code.

Reply:
Ah cool! Good to know